exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/xsort/ExternalSortBatch.java

```diff
@@ -514,13 +516,20 @@ public BatchGroup mergeAndSpill(LinkedList<BatchGroup> batchGroups) throws Schem
         outputContainer.setRecordCount(count);
         newGroup.addBatch(outputContainer);
       }
+      injector.injectChecked(context.getExecutionControls(), INTERRUPTION_WHILE_SPILLING, IOException.class);
       newGroup.closeOutputStream();
-      for (BatchGroup group : batchGroupList) {
-        group.cleanup();
-      }
-      hyperBatch.clear();
+      threw = false;
```

Comment:
`threw = false;` needs to be after any statement that could throw. I will add this in it's comment